static/scripts/chatLoggedin.js

```diff
@@ -35,33 +78,47 @@ function loadChatClient(session = null) {
 	document.head.appendChild(riotScript);
 }
 
+function hasActiveSessionInLocalStorage() {
+	return window.localStorage
+		&& window.localStorage.getItem('mx_hs_url')
+		&& window.localStorage.getItem('mx_access_token')
+		&& window.localStorage.getItem('mx_user_id');
+}
+
 function requestSession() {
 	return $.getJSON('/messenger/token');
 }
 
-function ready() {
-	if (window.innerWidth < 768) { // breakpoint: md
-		return; // screen to small to use embedded messenger
-	}
-
+async function initializeMessenger() {
 	// Find Matrix Session
-	// > in localstorage?
-	if (window.localStorage
-		&& window.localStorage.getItem('mx_hs_url')
-		&& window.localStorage.getItem('mx_access_token')
-		&& window.localStorage.getItem('mx_user_id')
-	) {
-		// session available, the chat will access it itself
-		loadChatClient();
+	let session;
+	if (hasActiveSessionInLocalStorage()) {
+		// session available, the messenger will access it itself
+		session = null;
 	} else {
-		// > get new Session from Server and pass it to the chat
-		requestSession()
-			.then(loadChatClient)
-			.catch((error) => {
-				/* eslint-disable-next-line no-console */
-				console.error('Failed to request Messenger Session.', error);
-			});
+		// get new session from Server
+		session = await requestSession();
 	}
+
+	addMatrixchatElement(session);
+	loadMessengerBundle();
+}
+
+let onReadyTriggered = false;
+
+async function onDocumentReady() {
+	if (window.innerWidth < 768) { // breakpoint: md
+		return false; // screen to small to use embedded messenger
```

Comment:
i would at least want to inform the user that a chat could be shown on bigger screens without showing a huge information on every page.. @dominikbruechner